sos/plugins/__init__.py

```diff
@@ -431,7 +431,7 @@ def _do_copy_path(self, srcpath, dest=None):
             self._copy_symlink(srcpath)
             return
         else:
-            if stat.S_ISDIR(st.st_mode):
+            if stat.S_ISDIR(st.st_mode) and os.access(srcpath, os.R_OK):
```

Comment:
This doesn't feel entirely right. What is _actually_ blocking the access here? Some part of the container compartmentalisation? On Linux root has always been able to bust through file system permissions and we rely on that to a fairly heavy extent - this change may have a side effect of causing us to not collect other data that we _can_ read simply because the permissions are either restrictive for some reason or broken. It is a **major** advantage for a tool like sos to be able to do this and to properly collect files even if the administrator has done something like `chmod 0000`.

Reply:
OK: I did some testing & discussed this with the Red Hat `sos` folks on a call today. Luckily it seems that `os.access()` uses `access(2)`, and that that appears to give the correct result, even if the root user is relying on being able to bust through `000` file modes.

I'll merge the patch today and we can keep a close eye for any problems in the time remaining to 3.6 (end of the week).